iohub/ndtiff.py

```diff
@@ -101,16 +189,18 @@ def str_channel_axis(self) -> bool:
         """Channel axis is string-valued"""
         return self._str_channel_axis
 
-    def _check_coordinates(
-        self, p: Union[int, str], t: int, c: Union[int, str], z: int
-    ):
+    @property
+    def ndtiff_axes(self) -> tuple[str]:
+        return ("position", "time", "channel", "z", "y", "x")
```

Comment:
Should this be a private property? 

NDTiff allows for arbitrary axis names. Does this mean that we are not supporting axis names outside of the ones defined here? That would be OK with me.

Reply:
I didn't know that these can be arbitrary strings. Then how can we support that if e.g. the channel axis is not named 'channel'?

Reply:
Yeah, NDTiff would allow you to create for example `("position", "time", "lf_channels", "fl_channels", "z", "y", "x")`. I think it's OK if we don't support that. I was mostly arguing for `ndtiff_axes` to be a private property, i.e. `_ndtiff_axes`. We could check that the ndtiff dataset axes are in `_ndtiff_axes` and throw a "not supported" error if that's not the case.

Reply:
https://github.com/czbiohub-sf/iohub/pull/185/commits/a6e13a8ae2a76d8e6e02afec4ad91d0debbe59f8